esmvalcore/preprocessor/_mask.py

```diff
@@ -119,28 +127,28 @@ def mask_landsea(cube, mask_out):
         try:
             fx_cube = cube.ancillary_variable('sea_area_fraction')
         except iris.exceptions.AncillaryVariableNotFoundError:
-            logger.debug('Ancillary variables land/sea area fraction not '
-                         'found in cube. Check fx_file availability.')
+            logger.debug(
+                "Ancillary variables land/sea area fraction not found in "
+                "cube. Check fx_file availability."
+            )
 
     if fx_cube:
-        fx_cube_data = da.broadcast_to(fx_cube.core_data(), cube.shape)
-        landsea_mask = _get_fx_mask(fx_cube_data, mask_out,
-                                    fx_cube.var_name)
-        cube.data = _apply_fx_mask(landsea_mask, cube.core_data())
+        fx_cube_data = np.broadcast_to(fx_cube.core_data(), cube.shape)
+        landsea_mask = _get_fx_mask(fx_cube_data, mask_out, fx_cube.var_name)
+        cube.data = _apply_mask(landsea_mask, cube.core_data())
```

Comment:
Maybe the `broadcast_to` operation could be moved into `_apply_mask` and then be supplied with a `chunks` argument so it uses the chunks that are aligned with the cube's data (perhaps by using `iris.util.broadcast_to_shape`)? Calling the function `_get_fx_mask` before broadcasting may be a bit more computationally efficient too.

Reply:
Great suggestion! Done inhttps://github.com/ESMValGroup/ESMValCore/pull/2515/commits/325db6db180f5249ecd7bc0d120419e95cea70b3

Reply:
:partying_face: